Add strategy index test to travis.

drvinceknight opened the pull request:
Closes #957. I am expecting this to fail on travis: two modules are not currently indexed (`hmm` and `memorytwo`). Using that as a test that this script works (once travis fails, will push the fix).

drvinceknight commented:
4738a96 has failed as expected:

```
 0.04s$ python run_strategy_indexer.py
memorytwo not in index
The command "python run_strategy_indexer.py" exited with 1.
```

Just pushing the fix.

marcharper commented on run_strategy_indexer.py:
```diff
@@ -0,0 +1,49 @@
+                 index_path="./docs/reference/all_strategies.rst",
+                 excluded=("_strategies", "__init__", "_filters", "human")):
+    """
+    Check if a module is contained in the index of strategies
```
Can we improve the docstring? Technically these aren't modules. Also let's use the standard style for parameters (just module_path is ok?) and the return value.

marcharper commented on run_strategy_indexer.py:
```diff
@@ -0,0 +1,49 @@
+    """
+    strategies_reference = read_index(index_path)
+    module_name = get_module_name(module_path)
+    if module_name not in excluded:
```
could use a single if (either way is ok)

marcharper commented on run_strategy_indexer.py:
```diff
@@ -0,0 +1,49 @@
+
+if __name__ == "__main__":
+
+    modules = glob.glob("./axelrod/strategies/*py")
```
"*.py" might be slightly more future proof

drvinceknight commented:
I think I've addressed your comments @marcharper. I've also added type hinting :)

meatballs commented:
I'm happy with this but I think an improvement might be to use the Pathlib module for file paths.

I don't think there's a problem here, but I've have had problems with file paths on windows before and Pathlib handles all that perfectly.

drvinceknight commented:
That's a good shout. I'll do that and push today :+1:

On Thu, 20 Apr 2017, 08:20 Owen Campbell, <notifications@github.com> wrote:

> I'm happy with this but I think an improvement might be to use the Pathlib
> module for file paths.
>
> I don't think there's a problem here, but I've have had problems with file
> paths on windows before and Pathlib handles all that perfectly.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Axelrod-Python/Axelrod/pull/980#issuecomment-295607606>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACCGWgPFUEipTyorDVLn58_7Y_ZNfONKks5rxwchgaJpZM4NB8PY>
> .
>

drvinceknight commented:
> but I've have had problems with file paths on windows before and Pathlib handles all that perfectly.

Great shout on Pathlib, it cleared up a lot of the code (no real point in 2 of the 3 functions as they're just simple method calls of the `Path` class). I'd seen it used before but never played around with it myself: it's lovely!.

Note: I removed the type specification in the docstring. This is mainly because I noticed just before pushing that it had become a lie (I changed the type hint to be a `Path` and didn't change the docstring). Seems sensible to me to remove the types from the docstrings when we have type hints in place but I'm happy to throw them back in for this function if we want them.

meatballs commented:
> Great shout on Pathlib, it cleared up a lot of the code (no real point in 2 of the 3 functions as they're just simple method calls of the Path class). I'd seen it used before but never played around with it myself: it's lovely!.

Yep. I'm a big fan!
